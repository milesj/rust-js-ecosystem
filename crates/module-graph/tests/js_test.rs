mod utils;

use starbase_sandbox::{assert_snapshot, create_sandbox};
use utils::*;

mod mjs {
    use super::*;

    #[test]
    fn export_star() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-star.mjs"
        ));
    }

    #[test]
    fn export_named() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-named.mjs"
        ));
    }

    #[test]
    fn export_default_function() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-def-func.mjs"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-def-anon-func.mjs"
        ));
    }

    #[test]
    fn export_default_class() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-def-class.mjs"
        ));
        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-def-anon-class.mjs"
        ));
    }

    #[test]
    fn export_default_reference() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/export-def-ref.mjs"
        ));
    }

    #[test]
    fn dynamic_import_top_level_await() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/dyn-import-tla.mjs"
        ));
    }

    #[test]
    fn dynamic_import_scope_depths() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/dyn-import-scopes.mjs"
        ));
    }

    #[test]
    fn dynamic_import_destructure_patterns() {
        let sandbox = create_sandbox("js");

        assert_snapshot!(generate_graph_for_file(
            sandbox.path(),
            "mjs/dyn-import-patterns.mjs"
        ));
    }
}
