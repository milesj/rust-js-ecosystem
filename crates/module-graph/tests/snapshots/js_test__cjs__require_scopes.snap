---
source: crates/module-graph/tests/js_test.rs
expression: "generate_graph_for_file(sandbox.path(), \"cjs/require-scopes.cjs\")"
---
{
    2: Module {
        exports: [
            Export {
                kind: Legacy,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 0,
                        end: 14,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Export {
                kind: Legacy,
                module_id: None,
                source: None,
                span: Some(
                    Span {
                        start: 22,
                        end: 36,
                    },
                ),
                symbols: [
                    ExportedSymbol {
                        kind: Value,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
        ],
        fragment: None,
        id: 2,
        imports: [],
        package_name: None,
        path: "/private/root/cjs/shared.cjs",
        query: None,
        source: JavaScriptModule {
            package_type: Cjs,
            source: "exports.number = 123;\nexports.string = \"abc\";\n",
            source_type: SourceType {
                language: JavaScript,
                module_kind: Module,
                variant: Jsx,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 0,
                exports_default: false,
                import_statements: 0,
                other_statements: 2,
                require_count: 0,
            },
        },
    },
    1: Module {
        exports: [],
        fragment: None,
        id: 1,
        imports: [
            Import {
                kind: SyncStatic,
                module_id: 2,
                source_request: "./shared.cjs",
                span: Span {
                    start: 0,
                    end: 23,
                },
                symbols: [],
                type_only: false,
            },
            Import {
                kind: SyncStatic,
                module_id: 2,
                source_request: "./shared.cjs",
                span: Span {
                    start: 56,
                    end: 79,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Namespace,
                        source_name: None,
                        symbol_id: None,
                        name: "ns",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: SyncStatic,
                module_id: 2,
                source_request: "./shared.cjs",
                span: Span {
                    start: 141,
                    end: 164,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "number",
                    },
                ],
                type_only: false,
            },
            Import {
                kind: SyncStatic,
                module_id: 2,
                source_request: "./shared.cjs",
                span: Span {
                    start: 222,
                    end: 245,
                },
                symbols: [
                    ImportedSymbol {
                        kind: Value,
                        source_name: None,
                        symbol_id: None,
                        name: "string",
                    },
                ],
                type_only: false,
            },
        ],
        package_name: None,
        path: "/root/cjs/require-scopes.cjs",
        query: None,
        source: JavaScriptModule {
            package_type: Cjs,
            source: "require(\"./shared.cjs\");\n\nfunction foo() {\n  const ns = require(\"./shared.cjs\");\n}\n\nfunction bar() {\n  return () => {\n    const { number } = require(\"./shared.cjs\");\n  };\n}\n\nclass Baz {\n  method() {\n    const { string } = require(\"./shared.cjs\");\n  }\n}\n",
            source_type: SourceType {
                language: JavaScript,
                module_kind: Module,
                variant: Jsx,
                always_strict: false,
            },
            stats: JavaScriptStats {
                dynamic_import_count: 0,
                export_statements: 0,
                exports_default: false,
                import_statements: 0,
                other_statements: 4,
                require_count: 4,
            },
        },
    },
}
